### ISSUE TITLE
new features if possible, mainly for use with google drive

### ISSUE THREAD

**Issue by devcrono** [cy33hc/ps4-ezremote-client#9]
hello @cy33hc your application is simply amazing and I would like to implement some things if possible

1 - that images are searched by CUSA and or that is presented if I had an image inside the folders in the games as an icon
2 - I would like to be able to map Team Shared Drive (shared drives) as I have one with many games there I could use my token and choose the drive ID
3 - connect to google directly without needing API
4 - work with services acount , same as rclone
5 - be able to use cloudflare works or g-index to map teamDrive

**Comment by devcrono** [comment]
If you want, send me the txt with the fields so I can translate it to pt-br and I'll send it to you right away

**Comment by cy33hc** [comment]
Use the English.ini https://github.com/cy33hc/ps4-ezremote-client/blob/master/data/assets/langs/English.ini for reference and update the existing Portuguese_BR.ini https://github.com/cy33hc/ps4-ezremote-client/blob/master/data/assets/langs/Portuguese_BR.ini

**Comment by devcrono** [comment]
hello  @cy33hc 
everything ready 100% translated.
taking advantage and looking at the file is just missing for google

1 - root_id = (in case the end of the link specifies the folder you wanted to share)
ex.(https://drive.google.com/folder/d/1WeE_u3RIc5wX-TDJywV1dxRgoIv5FYdO) in case you would only need this (1WeE_u3RIc5wX-TDJywV1dxRgoIv5FYdO)

2 - be able to use google services accounts, and be able to locate them as a.json file

OBS:I know all this just because of Rclone, take a look at it and I think you can implement it

https://rclone.org/drive/ 

[Portuguese_BR.zip](https://github.com/cy33hc/ps4-ezremote-client/files/11024799/Portuguese_BR.zip)

**Comment by cy33hc** [comment]
@devcrono I don't have shared drives with my google account. Can you test if shared drives work with this build. https://drive.google.com/file/d/1Yp2MWcMo8egiPyk8I1V1hjye_FZvxp1j/view?usp=sharing

**Comment by devcrono** [comment]
I managed to identify some errors

1 - now it presents in the "root" of the folder your files, the shared folders and the folders shared with you in the same place (see img2)

----------could put an option so that after he logs in he presents the folders
                              staff -> (when opening items from this directory)
                              shared drives -> (when opening items from that directory)
                              shared with me -> (when opening items from that directory)

and when you select the folders in question you would see their contents, this is to stay organized and beautiful

2 - even with the shared folders appearing however, it does not appear, nothing it appears zeroed, and with zero size too
example:
                                  ROOT FOLDER (content visible)
                                       xxxx.PHG (visible content)
                                 SHARED DRIVE (visible content)
                                       FOLDER A (invisible content)
                                                    YYYY.PKG (invisible content)

that is, it only appears until the shared drive and nothing else


3 - if by accident you press either side of the touchpad, the application closes

4 - you could put the ROOT_ID field below where you put your CLIENTE_ID and SENHA_ID (see the arrow on img1)

------------ in this field you could choose which folder you would like to automatically login to before login.

example (https://drive.google.com/folder/d/1WeE_u3RIc5wX-TDJywV1dxRgoIv5FYdO)

in case you would only need this (1WeE_u3RIc5wX-TDJywV1dxRgoIv5FYdO)
anyone with a google drive could write this code down and type

![img2](https://user-images.githubusercontent.com/69425591/227072075-e1a20525-8a05-4b7d-a317-d1faa81b39fa.jpg)
![img1](https://user-images.githubusercontent.com/69425591/227072078-df277b28-c9f6-4457-ad97-028eb3375f5b.jpg)

**Comment by devcrono** [comment]
> @devcrono Have you tried "Install from URL" function for direct links to package for goindex links? Links need to be unprotected.
yes I tried, the problem is that inside the root in folders and inside these folders the pkg content and they do not appear

OBS:if you can send me a DM so I can add you on telegram so we can try to improve your app, of course if you want

**Comment by cy33hc** [comment]
@devcrono I will be on vacation for 2 weeks. So I won't be looking into this until after holiday.

**Comment by devcrono** [comment]
> 

all right, happy holidays

**Comment by devcrono** [comment]
updating....  @cy33hc 
I generated a webdav repository via cloudflare with google index.
through this git

https://github.com/PBhadoo/google-drive-webdav-workers

and I noticed some things in the app *ps4-ezremote-client* the version that he sent me via Gdrive link, the connection is made but it doesn't show any content from any of the folders even though I changed to all types of servers (http,nginx, apache , etc...), placing https/http/webdav/dav/nothing/www any type of content is displayed

more if I use your old app *ps4-webdav-client*

connection is done, everything appears fine, but I have errors in the download, slow speeds, and especially if I put it to install directly before downloading, in the install function different from your new app, I get error messages in the console.

Would it be possible to see what is different between these 2 apps, in the webdav area, to find out why the new app cannot establish the connection?, and why the old app gives these automatic download errors, and slowness?

**Comment by cy33hc** [comment]
@devcrono In ezremote-zlient for webdav server you need prefix the url with webdav:// or webdavs:// instead of http:// or https://

**Comment by devcrono** [comment]
> @devcrono In ezremote-zlient for webdav server you need prefix the url with webdav:// or webdavs:// instead of http:// or https://

so there would be no way to put this function to recognize http/https because cloudflare only dispenses with the link that way if I try to modify it from the domain error

**Comment by cy33hc** [comment]
@devcrono No. you don't need change anything on webdav server side. Internally ezremote-client still uses http to communicate with the webdav server. The webdav:// prefix is just to indicate to ezremote-client that to use the webdav api for communication. In normal http server there is no PROPFIND web request, PROPFIND is an extension for specific for webdav.

**Comment by devcrono** [comment]
@cy33hc please check for me the option to use services account, and show the contents of shared drives folders

**Comment by cy33hc** [comment]
> updating.... @cy33hc I generated a webdav repository via cloudflare with google index. through this git
> 
> https://github.com/PBhadoo/google-drive-webdav-workers
> 
> and I noticed some things in the app _ps4-ezremote-client_ the version that he sent me via Gdrive link, the connection is made but it doesn't show any content from any of the folders even though I changed to all types of servers (http,nginx, apache , etc...), placing https/http/webdav/dav/nothing/www any type of content is displayed
> 
> more if I use your old app _ps4-webdav-client_
> 
> connection is done, everything appears fine, but I have errors in the download, slow speeds, and especially if I put it to install directly before downloading, in the install function different from your new app, I get error messages in the console.
> 
> Would it be possible to see what is different between these 2 apps, in the webdav area, to find out why the new app cannot establish the connection?, and why the old app gives these automatic download errors, and slowness?

There is a bug in the repo https://github.com/PBhadoo/google-drive-webdav-workers. In lines 468, 469 of worker.js, it's assigned wrong reversed.
```
                            'Content-Length': result.mimeType,
                            'Content-Type': result.size,
```
Should be
```
                            'Content-Length':  result.size,
                            'Content-Type': result.mimeType,
```

**Comment by cy33hc** [comment]
@devcrono Try this build to see if shared drives are listing content now. Also added option "Set Default Folder" to set the default folder after login to any site. For google drive, personal files are under the "My Drive" folder now to match google website.

https://drive.google.com/file/d/1Yp2MWcMo8egiPyk8I1V1hjye_FZvxp1j/view?usp=sharing

**Comment by devcrono** [comment]
> @devcrono Try this build to see if shared drives are listing content now. Also added option "Set Default Folder" to set the default folder after login to any site. For google drive, personal files are under the "My Drive" folder now to match google website.
> 
> https://drive.google.com/file/d/1Yp2MWcMo8egiPyk8I1V1hjye_FZvxp1j/view?usp=sharing

right in case then it has to appear
my drive -> files
shared drive -> files
and that?

please add option to locate .json file with services account.
which can be placed inside a pendrive and placed inside the root folder of the app.
with that any and all google download quota is removed

but I will test the new app

just install on top or remove the previous one?

**Comment by devcrono** [comment]
> > updating.... @cy33hc I generated a webdav repository via cloudflare with google index. through this git
> > https://github.com/PBhadoo/google-drive-webdav-workers
> > and I noticed some things in the app _ps4-ezremote-client_ the version that he sent me via Gdrive link, the connection is made but it doesn't show any content from any of the folders even though I changed to all types of servers (http,nginx, apache , etc...), placing https/http/webdav/dav/nothing/www any type of content is displayed
> > more if I use your old app _ps4-webdav-client_
> > connection is done, everything appears fine, but I have errors in the download, slow speeds, and especially if I put it to install directly before downloading, in the install function different from your new app, I get error messages in the console.
> > Would it be possible to see what is different between these 2 apps, in the webdav area, to find out why the new app cannot establish the connection?, and why the old app gives these automatic download errors, and slowness?
> 
> There is a bug in the repo https://github.com/PBhadoo/google-drive-webdav-workers. In lines 468, 469 of worker.js, it's assigned wrong reversed.
> 
> ```
>                             'Content-Length': result.mimeType,
>                             'Content-Type': result.size,
> ```
> 
> Should be
> 
> ```
>                             'Content-Length':  result.size,
>                             'Content-Type': result.mimeType,
> ```

dude, if in this script you can add the function of using google service accounts and remove the 300mb donwload limitation that I still haven't found, this script solves the problem of not having where to download the games from, you just have to do an index from google with their games, for a service account and that's it, all the games in the world can be downloaded.
there would only be a lack of images to be uploaded by the game cusa XD

**Comment by cy33hc** [comment]
@devcrono Just install over.

**Comment by cy33hc** [comment]
@devcrono I had created a shared drive for testing and I believe I have got shared drives functions working. Please test this version.  https://drive.google.com/file/d/1yuYaQ1ib00Y61yggBIJr0fENEktTG45R/view?usp=sharing

**Comment by devcrono** [comment]
> > @devcrono Try this build to see if shared drives are listing content now. Also added option "Set Default Folder" to set the default folder after login to any site. For google drive, personal files are under the "My Drive" folder now to match google website.
> > https://drive.google.com/file/d/1Yp2MWcMo8egiPyk8I1V1hjye_FZvxp1j/view?usp=sharing
> 
> right in case then it has to appear my drive -> files shared drive -> files and that?
> 
> please add option to locate .json file with services account. which can be placed inside a pendrive and placed inside the root folder of the app. with that any and all google download quota is removed
> 
> but I will test the new app
> 
> just install on top or remove the previous one?

you might not even believe it
but I was now testing the previous version and I will send you the following feed,
my drive works perfectly
shared with me also works
and shared folders (shared drives) does not work, it simply crashes the app.

Now I will test this new version.

taking advantage of the modification you made in the webdavS it also solved

**Comment by devcrono** [comment]
> @devcrono I had created a shared drive for testing and I believe I have got shared drives functions working. Please test this version. https://drive.google.com/file/d/1yuYaQ1ib00Y61yggBIJr0fENEktTG45R/view?usp=sharing

SIMPLY AMAZING........
Everything working perfectly..

Please now add the icons option, so when you enter the folder or something it will display the image, which the PKGView shows

**Comment by devcrono** [comment]
@cy33hc obrigado
Thanks 
The Google issue has been resolved

**Comment by devcrono** [comment]
> @devcrono I had created a shared drive for testing and I believe I have got shared drives functions working. Please test this version. https://drive.google.com/file/d/1yuYaQ1ib00Y61yggBIJr0fENEktTG45R/view?usp=sharing

one more thing 
Don't have a way to learn everything from your Google without needing an API?

I believe that if you can do it just by login, it would be easier for the end user.
Just enter your email and password

**Comment by devcrono** [comment]
> > @devcrono I had created a shared drive for testing and I believe I have got shared drives functions working. Please test this version. https://drive.google.com/file/d/1yuYaQ1ib00Y61yggBIJr0fENEktTG45R/view?usp=sharing
> 
> one more thing Don't have a way to learn everything from your Google without needing an API?
> 
> I believe that if you can do it just by login, it would be easier for the end user. Just enter your email and password

@cy33hc 
for example when you
do gdindex or gindex it already generates your client_id and secret_id for you
This repository does this as an example:

https://github.com/maple3142/GDIndex

  if it were possible to do it that way, the end user would only need to put his gmail address, the browser would try to validate and bring the folders

**Comment by cy33hc** [comment]
@devcrono Release 1.06. Added viewing of images and PKG file preview. https://github.com/cy33hc/ps4-ezremote-client/releases/tag/1.06

**Comment by devcrono** [comment]
I believe it is better to put the options button to be the general options button and in that menu put an option to close the application, X selects, options does the actions, O cancels everything, triangle mixed options, square view images 


Please check the possibility to make the app auto, validate GoogleDrive without needing API, using V3 or V4

**Comment by devcrono** [comment]
> @devcrono Release 1.06. Added viewing of images and PKG file preview. https://github.com/cy33hc/ps4-ezremote-client/releases/tag/1.06

try to do it that way create a mini square with the image of the CUSA in question before the name of the game or game folder e. when scrolling through the ps4 cursor or with the "track" selector in the X folder, the game image would expand, or else I would press square and the image would appear

**Comment by devcrono** [comment]
@cy33hc 
good afternoon.
doubt, are CDN, NFS, Samba, SMB protocols supported?

**Comment by cy33hc** [comment]
@devcrono Yes Samba/SMB is supported. Samba is the name of the software, SMB is the protocol. It's also same as Windows shares is also SMB protocol. NFS is not supported.

**Comment by devcrono** [comment]
@cy33hc about NFS I couldn't reproduce yet to test.
About the CDN, it is a way of distributing files on servers around the world. An example is cloudflare 

https://github.com/Gkiokan/hb-store-cdn-cli-server

See this repository, and a way to turn a share into a CDN, and use it in the HB store

Note: were you able to implement the question of making the image inside the folder, turn the icon before the name?

**Comment by devcrono** [comment]
@cy33hc 
Is there a way to add the function to show the drives, internal and external? or the moment you install it, it asks for the internal or external drive.

that is if you have an external drive of course

**Comment by devcrono** [comment]
@cy33hc 
another thing is the work in the background, if you put it to download and just press the ps button to download it, you should have a way for it to continue the download, it only cancels if you really clicked cancel

**Comment by devcrono** [comment]
@cy33hc 
another thing that can be done to make it even better, it was something that a friend came up with the idea of.
when you click to download or install it, it would take this pkg and divide it into multiple parts, like download managers do like orbit, it makes a large file turn into several small ones, which speeds up the connection.

it's possible?

**Comment by devcrono** [comment]
@cy33hc 
another suggestion that I believe would make your app more beautiful

on the left and right side appear listing of folders right?!

1 - leave only one side to always display the pkg information such as a photo, etc... just like you put when the person clicks with X, and on the other side the list of folders as it is today

imagine how lethal the pkg photo would appear on the left side. when you select it with square.

2 - place a button to see the log of all messages, and of everything that was done in the app, with errors, successes, etc... which one gave the error you just see the messages that one of them gave now which one?

3 - I think you should put the DELETE, EXCLUDE, CUT, etc... options in other colors (see better for example) because they are dangerous options.

in my case I don't want anyone to be able to delete games from my drive, you know, I could set a master password or something like that

4 - when clicking on INSTALL OR DOWNLOAD the app, ask where you want to install or download it, showing the external devices.

this would make the direct installation more dynamic, because I believe that 99% of the people who are using your wonderful app, don't download it, they install it directly.

5 - background, this is highly important, when clicking on the install button. the pkg already appears in the ps4 notifications, you could leave the app, go play and your game would continue downloading

that would be awesome.

6 - pause, resume download, download queue

wonderful

I'm here to help, I'll always bring ideas, and if you want I'm willing to help, my knowledge in C is very weak but I can help if you want, if it was in python maybe I could help you more

thanks and keep up the great work

**Comment by devcrono** [comment]
It is possible to make external links installed directly through the web interface. in version 1.09?

example

I take a mediafire link and paste it in the upload field. Would it download directly to the console?

**Comment by cy33hc** [comment]
> 4 - when clicking on INSTALL OR DOWNLOAD the app, ask where you want to install or download it, showing the external devices.
> 
> this would make the direct installation more dynamic, because I believe that 99% of the people who are using your wonderful app, don't download it, they install it directly.

Install from "Remote" will now install directly via embedded httpserver, without downloading first in v1.09.
